### DISCUSSION TITLE
Can I declare immutable parameter on AsyncNotifier class?

### DISCUSSION THREAD

**Discussion by devnta** [rrousselGit/riverpod#3494] [General]
**Sorry for my bad english.**
As title, Can I declare immutable parameter on AsyncNotifier class?
Ex:
```
class MyNotifier extends AsnycNotifier<String> {
final bool myParam = false;
......
Some code
}
```
`myParam` is immutable, no need rebuild when it changes, so I do not want declare it on State class.

**Comment by gdurandrexel** [comment]
Yes you can.

You can also declare mutable fields and change their value. A notifier is just a class. You can use it as you like. Just be aware of when a new instance is created and whether it may have an impact on the provided value.

**Reply by devnta** [reply]
Thanks for anwser.
In official document mentioned:
![image](https://github.com/rrousselGit/riverpod/assets/102949116/d62fffbf-f701-48cb-9f3a-97560b50d03d)
Can I declare TextEdittingController in AsyncNotifier class?
It's safe or not?
I might miss something....

**Comment by gdurandrexel** [comment]
I don't know about "safe", I am not familiar enough with flutter intrinsics.

You could first try the recommended solutions, using hooks or a StatefulConsumer, and have a look at [different ways to handle forms](https://github.com/rrousselGit/riverpod/discussions/3476).
If all that is not good for you, then you can try to use the TextEditingController in your provider. Maybe you won't be impacted by the egde-cases mentioned in the doc.
